watertap/unit_models/steam_heater_0D.py

```diff
@@ -0,0 +1,151 @@
+However, the constraints and properties defined in the IDAES unit model do not align with those in the available WaterTAP property packages.
+To address this, alternative constraints have been replaced to ensure full condensation based on the WaterTAP properties.
+Note that additional components like desuperheaters, drain mixers, and coolers are not included. If necessary, these can be modeled separately by adding heat exchangers and mixers to the flowsheet.
+To do: Consider incorporating as a modification to IDAES' FeedWaterHeater model directly in the IDAES repo"
+"""
+
+
+@declare_process_block_class(
+    "SteamHeater0D",
+    doc="""Feedwater Heater Condensing Section
+The feedwater heater condensing section model is a normal 0D heat exchanger
+model with an added constraint to calculate the steam flow such that the outlet
+of shell is a saturated liquid.""",
+)
+class SteamHeater0DData(HeatExchangerData):
+    CONFIG = HeatExchangerData.CONFIG()
+    CONFIG.declare(
+        "has_saturation_pressure_deviation",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Flag to indicate if saturation pressure deviation should be considered",
+            doc="""Indicates whether the saturation pressure deviation at the outlet of the steam heater should be
+               modeled. If True, 'saturation_pressure_deviation' needs to be specified by the user.""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.hot_side.config.property_package.component_list,
+            doc="Mass balance",
+        )
+        def outlet_liquid_mass_balance(b, t, j):
+            b.hot_side.properties_out[t].flow_mass_phase_comp["Vap", j].fix(0)
+            return (
+                b.hot_side.properties_in[t].flow_mass_phase_comp["Vap", j]
+                + b.hot_side.properties_in[t].flow_mass_phase_comp["Liq", j]
+                == b.hot_side.properties_out[t].flow_mass_phase_comp["Liq", j]
+            )
+
+        units_meta = (
+            self.config.hot_side.property_package.get_metadata().get_derived_units
+        )
+
+        self.saturation_pressure_deviation = Var(
+            self.flowsheet().time,
+            initialize=0,
+            units=units_meta("pressure"),
+            doc="Difference between the outlet pressure and the saturation pressure of the condensed steam",
+        )
+
+        if self.config.has_saturation_pressure_deviation:
+            self.saturation_pressure_deviation.unfix()
+        else:
+            self.saturation_pressure_deviation.fix()
+
+        @self.Constraint(self.flowsheet().time, doc="Saturation pressure constraint")
+        def outlet_pressure_sat(b, t):
+            return (
+                b.hot_side.properties_out[t].pressure
+                + b.saturation_pressure_deviation[t]
+                == b.hot_side.properties_out[t].pressure_sat
+            )
```

Comment:
Do you have any specific reasons for NOT using an inequality constraint here?

Reply:
@Morgan88888888 Thanks for the comment. I definitely agree with you that using an inequality constraint is a much better approach in this case. Specifically, in the MD flowsheet, I specify the steam pressure based on the assumed source—whether it's low, medium, or high pressure, along with the pressure value. Initially, I had not fixed the inlet pressure, opting instead to fix the saturation at the outlet to aid in the analysis. I've switched to using an inequality constraint and fixed the inlet pressure in my previous commit.